devimint/src/tests.rs

```diff
@@ -1884,6 +1884,214 @@ pub async fn test_offline_client_initialization(
+            .as_object()
+            .with_context(|| format!("No modules found in consensus config for peer {peer_id}"))?;
+
+        let client_consensus_modules = client_config["modules"]
+            .as_object()
+            .with_context(|| format!("No modules found in client config for peer {peer_id}"))?;
+
+        // Look for existing meta module to copy its configuration
+        let existing_meta_consensus = consensus_config_modules
+            .values()
+            .find(|module_config| module_config["kind"].as_str() == Some("meta"));
+
+        let existing_meta_client = client_consensus_modules
+            .values()
+            .find(|module_config| module_config["kind"].as_str() == Some("meta"));
+
+        let existing_meta_consensus = existing_meta_consensus
+            .with_context(|| {
+                format!("No existing meta module found in consensus config for peer {peer_id}")
+            })?
+            .clone();
+        let existing_meta_client = existing_meta_client
+            .with_context(|| {
+                format!("No existing meta module found in client config for peer {peer_id}")
+            })?
+            .clone();
+
+        // Find the highest existing module ID for the new module
+        let max_id = consensus_config_modules
+            .keys()
+            .chain(client_consensus_modules.keys())
+            .filter_map(|id| id.parse::<u32>().ok())
+            .max()
+            .unwrap_or(0);
+
+        let new_module_id = (max_id + 1).to_string();
+
+        info!(
+            "Adding new meta module with id {} for peer {} (copying existing meta module config)",
+            new_module_id, peer_id
+        );
+
+        // Add new meta module to consensus config by copying existing meta module
+        if let Some(modules) = consensus_config["modules"].as_object_mut() {
+            modules.insert(new_module_id.clone(), existing_meta_consensus);
+        }
+
+        // Add new meta module to client config by copying existing meta module
+        if let Some(modules) = client_config["modules"].as_object_mut() {
+            modules.insert(new_module_id.clone(), existing_meta_client);
+        }
+
+        // Write back the modified configs
+        let updated_consensus_content = serde_json::to_string_pretty(&consensus_config)
+            .with_context(|| format!("Failed to serialize consensus config for peer {peer_id}"))?;
+
+        fs::write(&consensus_config_path, updated_consensus_content)
+            .await
+            .with_context(|| format!("Failed to write consensus config for peer {peer_id}"))?;
+
+        let updated_client_content = serde_json::to_string_pretty(&client_config)
+            .with_context(|| format!("Failed to serialize client config for peer {peer_id}"))?;
+
+        fs::write(&client_config_path, updated_client_content)
+            .await
+            .with_context(|| format!("Failed to write client config for peer {peer_id}"))?;
```

Comment:
I'm pretty sure this one is never read from disk, so doesn't need patching really.